### ISSUE TITLE
EDMC.exe issue since last update

### ISSUE THREAD

**Issue by Eventure-TCE** [EDCD/EDMarketConnector#1511]
After updating EDMC to the latest version, I authorized it successfully at Frontier.

The GUI showing the success as usual, but any command of the EDMC.exe showing Invalid Credentials.

Here a screencap of EDMC.exe command:

![Unbenannt](https://user-images.githubusercontent.com/78804889/158876911-923dcd30-6fe0-47a0-9cdb-109e40698c15.JPG)

Here a screencap of EDMC GUI:
![Unbenannt2](https://user-images.githubusercontent.com/78804889/158876935-9f1f0992-17c5-4c4d-81bf-f38cfb72518b.JPG)

PS: Can you please tell me again, to what issue the error code 5 (System I/O) is related to? I sadly forgot it :) Thanks!

**Comment by Athanasius** [comment]
We've had reports of Frontie Auth/CAPI just being flaky since Update 11.

Anyway, what we *need* is the log files that you would have been asked for if you'd used the Bug Report template for this issue.  There'll be an `EDMC-debug.log` in the same place that https://github.com/EDCD/EDMarketConnector/wiki/Troubleshooting#debug-log-files talks about for the EDMarketConnector-debug.log.

It might also be helpful to run the command as `EDMC.exe --loglevel DEBUG -m test.csv` to get more information in that log file.

That command just worked perfectly fine for me, using a pure Frontier account (no Steam or Epic Games Store involved).

**Comment by Eventure-TCE** [comment]
Identified the problem:
EDMC not selecting the latest (new) journal file to read the commander name.

Here is the output part:
2022-03-17 19:25:38.239 UTC - DEBUG - 15452:7660:7660 EDMC.main:233: Using logfile "C:\Users\Hellfire\Saved Games\Frontier Developments\Elite Dangerous\Journal.220104094004.01.log

**Comment by Athanasius** [comment]
Oh, ffs, of *course* EDMC.py has its own damned copy/paste of the code for that.

I'll look at this tomorrow.

**Comment by Eventure-TCE** [comment]
How about the error code 5?

BTW, great job with the debug mode! Makes life much easier just in case!

**Comment by Athanasius** [comment]
> How about the error code 5?

I'm not seeing that in the screenshots and excerpts you've shared.  This is why I asked for you to attach the log file(s), because then I can see *everything* that might be relevant.

**Comment by Athanasius** [comment]
@Eventure-TCE Please test https://github.com/EDCD/EDMarketConnector/releases/tag/Release%2F5.3.4-beta1

**Comment by Eventure-TCE** [comment]
The issue with picking the wrong journal was fixed with the beta 1. Good job!

In connection with the error code 5, jonathan told me in the early days of EDMC that EDMC.exe outputs an exit code after executing a command.

Exit codes:
5 = System IO error
4 = not docked
3 = not verified
2 = login error
1 = server down
0 = success

With this I check what problem was detected by EDMC when a command was executed.

All are clear to me except exit code 5.
Since I don't have the problem myself, but a TCE user currently has it (Homunk), I was no longer familiar with what code 5 says so that I can help the user.

Should he execute the above command test "EDMC.exe --loglevel DEBUG -m test.csv" too to get the EDMarketConnector-debug.log? I would then like to post it here for you if you agree.

**Comment by Athanasius** [comment]
Oh, that.

There are currently two causes of `sys.exit(EXIT_SYS_ERR)` (which is `5`), and they'll both log an error:

```python
            except Exception:
                logger.exception("Can't read Journal file")
                sys.exit(EXIT_SYS_ERR)

            if not monitor.cmdr:
                logger.error('Not available while E:D is at the main menu')
                sys.exit(EXIT_SYS_ERR)
```

**Comment by Athanasius** [comment]
> Should he execute the above command test "EDMC.exe --loglevel DEBUG -m test.csv" too to get the EDMarketConnector-debug.log? I would then like to post it here for you if you agree.

They would need to run it that way in order to get a debug log, yes.  But note that it would be `EDMC-debug.log` as it's separate from the GUI EDMarketConnector.exe one.

**Comment by Athanasius** [comment]
@Eventure-TCE OK, so 5.3.4 is out.

But I'll make the query explicit.

Given currently there are **TWO** errors from EDMC.exe that use the same '5' (EXIT_SYS_ERR) code, would it not be preferable for us to change them to specific ones so you can tell the difference ?

What I'm proposing is to leave EXIT_SYS_ERR as `5`, but then add two more:

- EXIT_JOURNAL_READ_ERR = `6`
- EXIT_COMMANDER_UNKNOWN = `7`

Obivously you'd need to update *first* to handle either `5` or `6` for the one case and either `5` or `7` for the other.

**Comment by Eventure-TCE** [comment]
I agree and will add the new codes 6 and 7 to the next update of TCE.

**Comment by Athanasius** [comment]
Actually, now I look at that code, `7` is already `EXIT_ARGS`, so it'll be:

- EXIT_JOURNAL_READ_ERR = `7`
- EXIT_COMMANDER_UNKNOWN = `8`

**Comment by Athanasius** [comment]
And that change is now in just-released 5.4.0.
